tests/unit/insights/test_importer.py

```diff
@@ -1,34 +1,581 @@
+            ),
+        ]
+        candidates = [
+            ProductInsight(
+                barcode=DEFAULT_BARCODE,
+                type=InsightType.label,
+                value_tag="tag1",
+                id=uuid.UUID("5d71c235-2304-4473-ba1c-63f3569f44a0"),
+            ),
+            ProductInsight(
+                barcode=DEFAULT_BARCODE,
+                type=InsightType.label,
+                value_tag="tag2",
+                id=uuid.UUID("c984b252-fb31-41ea-b78e-6ca08b9f5e4b"),
+            ),
+        ]
+        (
+            to_create,
+            to_delete,
+        ) = InsightImporterWithIsConflictingInsight.get_insight_update(
+            candidates, references
+        )
+        assert to_create == [candidates[1]]
+        assert to_delete == [references[1]]
+
+    def test_generate_insights_no_predictions(self):
+        assert (
+            list(
+                InsightImporter.generate_insights(
+                    [],
+                    DEFAULT_SERVER_DOMAIN,
+                    automatic=True,
+                    product_store=FakeProductStore(),
+                )
+            )
+            == []
+        )
+
+    def test_generate_insights_missing_product_no_references(self, mocker):
+        get_existing_insight_mock = mocker.patch(
+            "robotoff.insights.importer.get_existing_insight", return_value=[]
+        )
+        assert (
+            list(
+                InsightImporter.generate_insights(
+                    [
+                        Prediction(
+                            type=PredictionType.category,
+                            barcode=DEFAULT_BARCODE,
+                            data={},
+                        )
+                    ],
+                    DEFAULT_SERVER_DOMAIN,
+                    automatic=True,
+                    product_store=FakeProductStore(),
+                )
+            )
+            == []
+        )
+        get_existing_insight_mock.assert_called_once()
+
+    def test_generate_insights_missing_product_with_reference(self, mocker):
+        reference = ProductInsight(barcode=DEFAULT_BARCODE, type=InsightType.category)
+        get_existing_insight_mock = mocker.patch(
+            "robotoff.insights.importer.get_existing_insight",
+            return_value=[reference],
+        )
+        generated = list(
+            InsightImporter.generate_insights(
+                [
+                    Prediction(
+                        type=PredictionType.category,
+                        barcode=DEFAULT_BARCODE,
+                        data={},
+                    )
+                ],
+                DEFAULT_SERVER_DOMAIN,
+                automatic=True,
+                product_store=FakeProductStore(),
+            )
+        )
+        assert generated == [([], [reference])]
+        get_existing_insight_mock.assert_called_once()
+
+    def test_generate_insights_missing_product_creation_and_deletion(self, mocker):
```

Comment:
I think a docstring would be welcome, because I'm not sure what you are testing here !

(missing applies to insights, not to product ?)

Reply:
`missing` shouldn't be here at all, copy-pasting unit tests is the culprit here 😄 
It's fixed! (I've added a docstring too)